x-pack/plugins/apm/server/lib/services/get_service_transaction_group_comparison_statistics.ts

```diff
@@ -139,44 +142,125 @@ export async function getServiceTransactionGroupComparisonStatistics({
+          }),
+        }));
+        const throughput = bucket.timeseries.buckets.map(
+          (timeseriesBucket) => ({
             x: timeseriesBucket.key,
-            y: getLatencyValue({
-              latencyAggregationType,
-              aggregation: timeseriesBucket.latency,
-            }),
-          }));
-          const throughput = bucket.timeseries.buckets.map(
-            (timeseriesBucket) => ({
-              x: timeseriesBucket.key,
-              y: timeseriesBucket.throughput_rate.value,
-            })
-          );
-          const errorRate = bucket.timeseries.buckets.map(
-            (timeseriesBucket) => ({
-              x: timeseriesBucket.key,
-              y: calculateTransactionErrorPercentage(
-                timeseriesBucket[EVENT_OUTCOME]
-              ),
-            })
-          );
-          const transactionGroupTotalDuration =
-            bucket.transaction_group_total_duration.value || 0;
-          return {
-            transactionName,
-            latency,
-            throughput,
-            errorRate,
-            impact: totalDuration
-              ? (transactionGroupTotalDuration * 100) / totalDuration
-              : 0,
-          };
-        }),
-        'transactionName'
-      );
+            y: timeseriesBucket.throughput_rate.value,
+          })
+        );
+        const errorRate = bucket.timeseries.buckets.map((timeseriesBucket) => ({
+          x: timeseriesBucket.key,
+          y: calculateTransactionErrorPercentage(
+            timeseriesBucket[EVENT_OUTCOME]
+          ),
+        }));
+        const transactionGroupTotalDuration =
+          bucket.transaction_group_total_duration.value || 0;
+        return {
+          transactionName,
+          latency,
+          throughput,
+          errorRate,
+          impact: totalDuration
+            ? (transactionGroupTotalDuration * 100) / totalDuration
+            : 0,
+        };
+      });
     }
   );
 }
+
+export async function getServiceTransactionGroupComparisonStatisticsPeriods({
+  serviceName,
+  transactionNames,
+  setup,
+  numBuckets,
+  searchAggregatedTransactions,
+  transactionType,
+  latencyAggregationType,
+  comparisonStart,
+  comparisonEnd,
+  environment,
+}: {
+  serviceName: string;
+  transactionNames: string[];
+  setup: Setup & SetupTimeRange;
+  numBuckets: number;
+  searchAggregatedTransactions: boolean;
+  transactionType: string;
+  latencyAggregationType: LatencyAggregationType;
+  comparisonStart?: number;
+  comparisonEnd?: number;
+  environment?: string;
+}) {
+  const { start, end } = setup;
+
+  const commomProps = {
+    setup,
+    serviceName,
+    transactionNames,
+    searchAggregatedTransactions,
+    transactionType,
+    numBuckets,
+    latencyAggregationType: latencyAggregationType as LatencyAggregationType,
+    environment,
+  };
+
+  const currentPeriodPromise = getServiceTransactionGroupComparisonStatistics({
+    ...commomProps,
+    start,
+    end,
+  });
```

Comment:
any reason why this needs an intermediate variable, rather than inlining it in `await Promise.all()`?